templates/.travis.yml.mustache

```diff
@@ -1,47 +1,64 @@
+      "
+  script:
+  - echo -e "${ANSI_YELLOW}Building ${CONTRIB_NAME}...${ANSI_RESET}" && echo -en 'travis_fold:start:script\\r'
+  - |
+    docker exec COQ /bin/bash --login -c "
+      export PS4='+ \e[33;1m(\$0 @ line \$LINENO) \$\e[0m '
+      set -ex
+      sudo chown -R coq:coq /home/coq/${CONTRIB_NAME}
+      opam install ${CONTRIB_NAME} -v -y -j ${NJOBS}
+      "
+  - docker stop COQ  # optional
+  - echo -en 'travis_fold:end:script\\r'
 
-script:
-- nix-build --argstr coq-version-or-url "$COQ" --extra-substituters https://coq.cachix.org --trusted-public-keys "cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= coq.cachix.org-1:5QW/wwEnD+l2jvN6QRbRRsa4hBHG3QiQQ26cxu1F5tI="
+nix: &NIX
+  language: nix
+  script:
+  - nix-build --argstr coq-version-or-url "$COQ" --extra-substituters https://coq.cachix.org --trusted-public-keys "cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= coq.cachix.org-1:5QW/wwEnD+l2jvN6QRbRRsa4hBHG3QiQQ26cxu1F5tI="
 
 matrix:
   include:
 
-  # Test supported versions of Coq
-{{# tested_coq_versions }}
-  - env: COQ={{ version_or_url }}
-{{/ tested_coq_versions }}
+  # Test supported versions of Coq via Nix
+{{# tested_coq_nix_versions }}
+  - env:
+    - COQ={{ version_or_url }}
+    <<: *NIX
+{{/ tested_coq_nix_versions }}
 
-  # Test opam package
-  - language: minimal
-    sudo: required
-    services: docker
-    env:
-    - COQ_IMAGE=coqorg/coq:{{ tested_coq_opam_version }}
-    - CONTRIB_NAME={{ shortname }}
+  # Test supported versions of Coq via OPAM
+{{# tested_coq_opam_versions }}
+  - env:
+    - COQ_IMAGE=coqorg/coq:{{ version }}
+    - CONTRIB_NAME=coq-{{ shortname }}
     - NJOBS=2
-    install: |
-      # Prepare the COQ container
-      docker pull ${COQ_IMAGE}
-      docker run -d -i --init --name=COQ -v ${TRAVIS_BUILD_DIR}:/home/coq/${CONTRIB_NAME} -w /home/coq/${CONTRIB_NAME} ${COQ_IMAGE}
-      docker exec COQ /bin/bash --login -c "
-        # This bash script is double-quoted to interpolate Travis CI env vars:
-        echo \"Build triggered by ${TRAVIS_EVENT_TYPE}\"
-        export PS4='+ \e[33;1m(\$0 @ line \$LINENO) \$\e[0m '
-        set -ex  # -e = exit on failure; -x = trace for debug
-        opam update -y
-        opam install -y -j ${NJOBS} --deps-only .
-        opam config list
-        opam repo list
-        opam list
-        "
-    script:
-    - echo -e "${ANSI_YELLOW}Building ${CONTRIB_NAME}...${ANSI_RESET}" && echo -en 'travis_fold:start:script\\r'
-    - |
-      docker exec COQ /bin/bash --login -c "
-        export PS4='+ \e[33;1m(\$0 @ line \$LINENO) \$\e[0m '
-        set -ex
-        sudo chown -R coq:coq /home/coq/${CONTRIB_NAME}
-        opam install -v -y -j ${NJOBS} .
-        "
-    - docker stop COQ  # optional
-    - echo -en 'travis_fold:end:script\\r'
+    <<: *OPAM
+{{/ tested_coq_opam_versions }}
+
+{{# extracted }}
+  # Test extracted via OPAM
+  - env:
+    - COQ_IMAGE=coqorg/coq:{{ tested_extracted_coq_opam_version }}
```

Comment:
Note that this could be a list as well.

Reply:
Indeed it could be a list, but in that case there's no simple way to apply the template for the OPAM file `extracted.opam.mustache` to get a valid result when the list has several entries. The multiple OPAM files would have to be generated programmatically, which you already suggested above. I think this is a good idea and will likely look at it soon, but I will merge this for now.